mods/default/functions.lua

```diff
@@ -293,12 +293,26 @@ minetest.register_abm({
 -- Dig upwards
 --
 
-function default.dig_up(pos, node, digger)
+local in_dig_up = false
+local mapgen_limit = tonumber(minetest.settings:get("mapgen_limit")) or 31007
+function default.dig_up(pos, node, digger, max_height)
+	if in_dig_up then return end -- Avoid excess calls
 	if digger == nil then return end
-	local np = {x = pos.x, y = pos.y + 1, z = pos.z}
-	local nn = minetest.get_node(np)
-	if nn.name == node.name then
-		minetest.node_dig(np, nn, digger)
+	max_height = max_height or mapgen_limit
+
+	for y = pos.y + 1, math.min(pos.y + max_height, mapgen_limit) do
+		local up_pos  = { x = pos.x, y = y, z = pos.z}
```

Comment:
`vector.new(pos.x, y, pos.z)`

Reply:
Using the vector object results in a performance tradeoff (negligible though), and I don't see the need of the helpers.

Reply:
while that's true this goes against explicit recommendation of the docs:
https://github.com/minetest/minetest/blob/master/doc/lua_api.md#compatibility-notes

Reply:
Oh my. Gotta do it then.